examples/Spirit/spirit_utils.cc

```diff
@@ -170,11 +176,222 @@ std::tuple<
 //   uint16_t knotpoints, // Number of knot points per mode
 //   double mu = 1){
 //   int numModes = modeSeqMat.rows(); 
-//   Eigen::VectorXi knotpointMat = Eigen::MatrixXi::Constant(numModes,1,knotpoints);
-//   return createSpiritModeSequence(plant, modeSeqMat,knotpointMat, mu);
+//   std::vector<int> knotpointVect = Eigen::MatrixXi::Constant(numModes,1,knotpoints);
+//   return createSpiritModeSequence(plant, modeSeqMat,knotpointVect, mu);
 // }
 
 
+
+
+
+
+
+
+
+
+
+// **********************************************************
+//   SETSPIRITJOINTLIMITS 
+
+
+template <typename T> 
+void setSpiritJointLimits(drake::multibody::MultibodyPlant<T> & plant, 
+                          dairlib::systems::trajectory_optimization::Dircon<T>& trajopt,  
+                          int iJoint, 
+                          double minVal, 
+                          double maxVal  ){
+
+  auto positions_map = multibody::makeNameToPositionsMap(plant);
+  auto velocities_map = multibody::makeNameToVelocitiesMap(plant);
+  // std::cout<<"joint_" + std::to_string(iJoint)<<std::endl;
+  int N_knotpoints = trajopt.N();
+  for(int i = 0; i<N_knotpoints;i++){
+    auto xi = trajopt.state(i);
+    trajopt.AddBoundingBoxConstraint(minVal,maxVal,xi(positions_map.at("joint_" + std::to_string(iJoint))));
+  }
+}
+
+template <typename T> 
+void setSpiritJointLimits(drake::multibody::MultibodyPlant<T> & plant, 
+                          dairlib::systems::trajectory_optimization::Dircon<T>& trajopt,  
+                          std::vector<int> iJoints, 
+                          std::vector<double> minVals, 
+                          std::vector<double> maxVals  ){
+  int numJoints = iJoints.size();
+  assert(numJoints==minVals.size());
+  assert(numJoints==maxVals.size());
+  for (int i = 0; i<numJoints; i++){
+    setSpiritJointLimits( plant, 
+                          trajopt,  
+                          iJoints[i], 
+                          minVals[i], 
+                          maxVals[i] );
+  }
+}
+template <typename T> 
+void setSpiritJointLimits(drake::multibody::MultibodyPlant<T> & plant, 
+                          dairlib::systems::trajectory_optimization::Dircon<T>& trajopt,  
+                          std::vector<int> iJoints, 
+                          double minVal, 
+                          double maxVal  ){
+
+  for (int iJoint: iJoints){
+    setSpiritJointLimits( plant, 
+                          trajopt,  
+                          iJoint, 
+                          minVal, 
+                          maxVal );
+  }
+}
+template <typename T> 
+void setSpiritJointLimits(drake::multibody::MultibodyPlant<T> & plant, 
+                          dairlib::systems::trajectory_optimization::Dircon<T>& trajopt ){
+  // Upper doesn't need a joint limit for now, but we may eventually want to
+  // change this to help the optimizations
```

Comment:
Why does it not need a joint limit? Does it not have one? Either way if it haves one we should add it here and now.

Reply:
I think the limit would be something like -2*pi to 2*pi, not sure how to handle this in this framework or if we even can.